src/EnergyPlus/SizingManager.cc

```diff
@@ -2135,7 +2204,7 @@ void DetermineSystemPopulationDiversity(EnergyPlusData &state)
                     int SysSizNum = UtilityRoutines::FindItemInList(
                         FinalSysSizing(AirLoopNum).AirPriLoopName, state.dataSize->SysSizInput, &SystemSizingInputData::AirPriLoopName);
                     if (SysSizNum == 0) SysSizNum = 1; // use first when none applicable
-                    if (FinalSysSizing(AirLoopNum).OAAutoSized) {
+                    if (FinalSysSizing(AirLoopNum).OAAutoSized && SysSizInput(SysSizNum).OccupantDiversity == AutoSize) {
```

Comment:
For this logic change, when comparing the existing logic and the new one, I feel there could be a pitfall: 

If we give the new software to a use who is still modeling some traditional stuff and used the OAAutoSized option--not related to any of the the new SP option or controller at all; but then he also left some value to the OccupantDensity input field when exploring the new field out (and not setting the `D` field back to the default auto-size again); then according to the new logic, this block will be not be executed---which is be unexpected--actually this block need to be executed in this scenario even if the `D` is accidentally inputted with some value (rather than being left as “autosize”).

Probably here some criterion based on a more concrete "new method selection"---that selection the option that actively using the Occupant Density" should be used as the criterion, rather than the single `D` auto-size to avoid the potential pitfall above.

Reply:
I'm not sure that I follow the description of the pitfall. 

The default value for the new field is `autosize` which keeps the calculation of `D` the same as before (based on schedules and design occupancy levels). If a user chooses to set a value for D, that value instead and the loop to determine the maximum concurrent occupancy is skipped. This occurs whether or not the SP is used, and that was the proposed intent.

Are you suggesting that the user-specified `D` be only used if the SP is selected?

Reply:
Actually I am talking about a scenario that the user is using the "new" software but modeling the old options--suppose that he is not using any of the new features (e.g. simple procedure, nor the new controller option) developed in this branch; and his settings makes that `FinalSysSizing(AirLoopNum).OAAutoSized)` becomes true. But also, he pokes around and (say accidently) changes the new "OccupantDiveristy" input field (since `D` is now an input option in the new software) to some explicit value (e.g. 0.5) and does not change that back to "autosize". 

Then according to the new logic, the block under this `if` would not be executed because `SysSizInput(SysSizNum).OccupantDiversity == AutoSize` become false. However, in his model, this `D` input really does not matter (not matter it is 0.5 or "autosize"), and the block should have been executed based on his original simulation choice.

Reply:
If in that example `D` doesn't matter, why would we want run that code block if the input field is not `autosize`? Unless I'm missing something, that code block is only used to calculate `D` prior to the actual full annual simulation. So if `D` doesn't matter, it seems like bypassing it if the user specifies a value for it would save some computing time. Am I missing your point?

Reply:
I think the principle applied here is that "when we simulation the old options, the new program and the old program should do the same thing--say execute the same block." So here my expectation is that if I am simulating the old options, the code should behave the same way as the old code do--but this is not the case in the new code since now you have `D` as an available input. In the examples above, I am trying to make an imaginary scenario that it will break this--where the new code might yield different execution route. 

Now if you look at the original code and your code side by side: the only difference is the `D`-auto-size criterion is now added. In a simulation that we simulate the old option, suppose that the user's input will get `FinalSysSizing(AirLoopNum).OAAutoSized` to be true; then it will execute the `if` block. 

But in your new code, even `FinalSysSizing(AirLoopNum).OAAutoSized` is true, the block will still dependent on whether `D`-autosize is true. If `D` is autosize (by default, good), then the block will be executed, which will match the old code execution order; 

However, in case `D` is accidentally not `autosize` but an explicitly input value (e.g. 0.5, accidentally given by user playing around that input), then your new code will not execute the `if` block--actually this is not expected, as it yields a different result with the old code. The reason why I think called it is "accidental" is because this is not a problem in the old code as there is no such `D` input; but it is possible to fail in your new code, as `D` is an available input, and an user can accidentally assigned a value to it when playing around. 

So the new code need to make sure that when I am simulation old options, the results are not affected by how you (accidentally or whatever) inputted the `D` value---no mater it is 1) left untouched as the "autosize" default, or 2)it was accidentally inputted by a user an explicit value. My example tries to demonstrate that when simulating the old options, your current new code will yield different execution routes at the `if` point when you select between 1) and 2), which could very likely yield different simulation results---and that's why I consider as this as a potential pitfall.

Reply:
@lymereJ Sorry for the confusion, I had thought that `D` was a completely new variable added. I looked through some more codes line and realized that the `D` probably could be considered always "autosized" in the old code; but now in this new feature development it is made manually adjustable. 

I think if this the correct understanding on the design intent of your new feature, then your new logic is perfectly fine. Sorry for the previous incorrect interpretation.

Reply:
Oh, I see, yes, that's the correct understanding. No worries, glad this is sorted out!